cranelift/wasm/src/code_translator.rs

```diff
@@ -2407,8 +2407,16 @@ fn translate_store<FE: FuncEnvironment + ?Sized>(
     state: &mut FuncTranslationState,
     environ: &mut FE,
 ) -> WasmResult<()> {
-    let val = state.pop1();
-    let val_ty = builder.func.dfg.value_type(val);
+    let mut val = state.pop1();
+    let mut val_ty = builder.func.dfg.value_type(val);
+
+    // Boolean-vector types don't validate with a `store` instruction, so
+    // bitcast them to a vector type which is compatible with the store
+    // instruction.
+    if val_ty.is_vector() && val_ty.lane_type().is_bool() {
+        val = builder.ins().raw_bitcast(I8X16, val);
```

Comment:
Should we instead define that boolean vectors can be stored using `store` directly and are stored as all-zero or all-one lanes?

Reply:
That is a good approach but I don't mind what @alexcrichton tried here because:
 - vector translation already involves a lot of bitcasting so this doesn't make the problem worse; if we want to solve the bitcasting we should just solve all of and close #1147
 - changing `load` and `store` to allow boolean vectors but not boolean scalars seems like it might cause errors somewhere; this fix would immediately fix a fuzz bug
 - I wouldn't mind settling the general issue about `load`s and `store`s with booleans once for all: I've seen mention recently of setting a memory representation for boolean scalars... If we settle it once for all (do we have an issue for this?) and document it then it might make sense to avoid the bitcast (if that's what is decided).

Reply:
I just created #3205 for this. Agreed that for now, the immediate fuzzbug fix is fine.

Reply:
Ok I'm gonna go ahead and merge this fix for now and defer to #3205 for future updates.